Fixed Issue #25126:Newletter subscription issue

rahulwebkul opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento/magento2#25126: Newletter subscription issue

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Go to any page on frontend and find newsletter section at bottom right of the page.
2. Login into admin panel.
3. Disable Newsletter. Navigate to Store -> Configuration -> Customer -> Newsletter -> General Options and set enable to **NO**
4. Now go to opened page on frontend and try to subscribe.
5. Now there will be no route page.

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

Commit:
Issue #25126 fixed

Commit:
Fixed annotaion issue in Subscribe.php Controller

Fixed annotaion issue in Subscribe.php Controller

rahulwebkul commented:
@melnikovi @buskamuza - please review this pull request

rahulwebkul commented:
@melnikovi @buskamuza - Is there any changes required from my end for this pull request. Please let me know so that i can update it accordingly.

melnikovi commented:
I think it might be good idea to hide form as well, need to confirm with PO. I will follow up with community engineering team.

melnikovi commented on app/code/Magento/Newsletter/Controller/Subscriber.php:
```diff
@@ -41,24 +49,67 @@ abstract class Subscriber extends \Magento\Framework\App\Action\Action
+    /**
+     * @var ScopeConfigInterface
+     */
+    protected $_scopeConfig;
```
Properties should be private if there are no reasons to make them protected. Also, by naming convention we don't use underscore in naming.

rahulwebkul replied:
@melnikovi - I have updated the variable name and updated the scope of possible variables.

Commit:
varible name and scope updated

Commit:
annotation issue fixed

Commit:
Update NewAction.php

rahulwebkul commented:
@melnikovi - Hi! regarding Integration Tests
I try to run integration testing in my local instance. Test was successful. I tried 2 times.
But here it is showing failed integration test .
Can you let me know the environment details so that i can check this.?
Or can you help me with integration test?

Commit:
Update NewAction.php

rahulwebkul commented:
@melnikovi - I have implement the requested changes. Please review pull request and let me know there is any update required from my end.

rodrigowebjump reviewed (changes requested):
Hi @rahulwebkul Thanks for your PR.

Please revert refactor of properties declaration. 

Try to keep the changes as minimum as possible. 
As you can see changing this properties forces you to refactor many class of core and also perhaps my third part modules will became incompatible.

Your PR should fix specifically the check if feature is enabled in admin.

Thanks

Commit:
code updated

Commit:
code reverted to original code

rahulwebkul commented:
@rodrigowebjump - I updated the other files due to some miscommunication between me and @melnikovi .
Now i have updated the only file and restored the other files to its original state.
I have also added backward compatibility. So there will be no issue in child classes. 
Please check and approve the changes.

VladimirZaets commented on app/code/Magento/Newsletter/Controller/Subscriber.php:
```diff
@@ -3,20 +3,29 @@
+    /**
+     * Configuration path to newsletter active setting
+     */
+    const XML_PATH_NEWSLETTER_ACTIVE = 'newsletter/general/active';
```
Please add scope to constant. In this case it should be private

rahulwebkul replied:
@VladimirZaets - I have added the scope in constant. Please check.

VladimirZaets reviewed (changes requested):
Please take a look at my review comments

Commit:
Update Subscriber.php

VladimirZaets commented:
Hi @rahulwebkul . Thanks for collaboration. Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. Please cover your fix by automated tests.

engcom-Charlie commented:
I'll take care of automated tests.

engcom-Charlie committed:
test coverage, revert files permissions

engcom-Charlie committed:
Merge branch '2.4-develop' into issue-25126

# Conflicts:
#	dev/tests/integration/testsuite/Magento/Newsletter/Controller/SubscriberTest.php

engcom-Charlie committed:
merge conflicts, fix static

rodrigowebjump commented:
@magento give me test instance

magento-engcom-team commented:
Hi @rodrigowebjump. Thank you for your request. I'm working on Magento instance for you

magento-engcom-team commented:
Hi @rodrigowebjump, here is your [new Magento instance](https://pr-25127.instances.magento-community.engineering).
Admin access: [https://pr-25127.instances.magento-community.engineering/admin_d41f](https://pr-25127.instances.magento-community.engineering/admin_d41f)
Login: `8af943a5` Password: `5361fb5d2244`
Instance will be terminated in up to 3 hours.

engcom-Charlie committed:
Merge branch '2.4-develop' into issue-25126

# Conflicts:
#	dev/tests/integration/testsuite/Magento/Newsletter/Controller/SubscriberTest.php

lenaorobei commented on app/code/Magento/Newsletter/Controller/Subscriber.php:
```diff
@@ -3,20 +3,29 @@
+    /**
+     * Configuration path to newsletter active setting
+     */
+    private const XML_PATH_NEWSLETTER_ACTIVE = 'newsletter/general/active';
```
Please see `\Magento\Newsletter\Model\Config::isActive`.

lenaorobei reviewed (commented):
Please update the feature branch and see review comment.

lenaorobei commented:
Closing this PR due to inactivity. Please feel free to reopen if you are going to continue working on it.
Thank you.
